controller/linodecluster_controller.go

```diff
@@ -192,6 +210,34 @@ func (r *LinodeClusterReconciler) reconcile(
 	return res, nil
 }
 
+func (r *LinodeClusterReconciler) reconcilePreflightLinodeVPCCheck(ctx context.Context, logger logr.Logger, clusterScope *scope.ClusterScope) (ctrl.Result, error) {
+	name := clusterScope.LinodeCluster.Spec.VPCRef.Name
+	namespace := clusterScope.LinodeCluster.Spec.VPCRef.Namespace
+	if namespace == "" {
+		namespace = clusterScope.LinodeCluster.Namespace
+	}
+	linodeVPC := infrav1alpha2.LinodeVPC{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+	}
+	if err := clusterScope.Client.Get(ctx, client.ObjectKeyFromObject(&linodeVPC), &linodeVPC); err != nil {
+		logger.Error(err, "Failed to fetch LinodeVPC")
+		if reconciler.RecordDecayingCondition(clusterScope.LinodeCluster,
+			ConditionPreflightLinodeVPCReady, string(cerrs.CreateClusterError), err.Error(),
+			reconciler.DefaultTimeout(r.ReconcileTimeout, reconciler.DefaultClusterControllerReconcileTimeout)) {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{RequeueAfter: reconciler.DefaultClusterControllerReconcileDelay}, nil
+	} else if !linodeVPC.Status.Ready || linodeVPC.Spec.VPCID == nil {
```

Comment:
Is it possible to have the status be Ready with a nil VPC ID?

Reply:
I was re-using existing logic in other parts of capl-controller but I think it makes sense to just check for either statur of vpcid. I've updated the code to just check for status to be ready.